x/ecocredit/server/marketplace/buy.go

```diff
@@ -250,10 +175,20 @@ func (k Keeper) fillOrder(ctx context.Context, sellOrder *api.SellOrder, buyerAc
 
 // getTotalCost calculates the cost of the order by multiplying the price per credit, and the amount of credits
 // desired in the order.
-func getTotalCost(pricePerCredit sdk.Int, amtCredits string) (sdk.Int, error) {
-	amtCreditsInt, ok := sdk.NewIntFromString(amtCredits)
-	if !ok {
-		return sdk.Int{}, fmt.Errorf("could not convert %s to %T", amtCreditsInt, sdk.Int{})
+func getTotalCost(pricePerCredit sdk.Int, amtCredits math.Dec) (sdk.Int, error) {
+	amountDec, err := amtCredits.Mul(math.NewDecFromInt64(pricePerCredit.Int64()))
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	amountDec, err = amountDec.QuoInteger(math.NewDecFromInt64(1))
```

Comment:
it's to eliminate anything to the right of the decimal. otherwise, cannot convert to `sdk.Int`

Reply:
hmm, I thought that you want to convert `{1.001, exponent = 4}` to `Int(10010)`. Why cutting the decimal places?

Reply:
so the bulk of that is already done in the lines above the reviewed snippet:

```go
	amountDec, err := amtCredits.Mul(multiplier)
	if err != nil {
		return sdk.Int{}, err
	}
```

however, in cases of small costs, you can end up with remaining decimals, which need to be eliminated, otherwise you can't put it in `sdk.Int`. 

example:
`quantity of credits to purchase: 10.54321`
`Price per credit: 30uatom`

`Total Cost: 316.2963`

i dont think this would happen too often as 30uatom is a suuuuuuper tiny amt to pay for credits but i guess it could happen with user input error. 

only question left is what to do in these cases: trim decimal? error out? something else?

Reply:
I see. Well, there is no smaller unit than `uatom`. So you can't pay `0.1` uatom. We should either trim decimal or round it. I prefer trimming.

Could you add comments next to the code?

Is there any other way to cut decimals? It feels to me that the `Quo` is suboptimal. Maybe we can create a new task to look at it to not block this one?